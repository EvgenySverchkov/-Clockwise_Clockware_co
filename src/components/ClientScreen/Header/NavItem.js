import React from "react";
import { Link } from "react-router-dom";

import PropTypes from "prop-types";

const NavItem = ({title, link}) => (
    <li className="nav-item">
        <Link className="nav-link" to={link}>
            {title}
        </Link>
    </li>
);

NavItem.propTypes = {
    title: PropTypes.string,
    link: PropTypes.string
}

export default NavItem;