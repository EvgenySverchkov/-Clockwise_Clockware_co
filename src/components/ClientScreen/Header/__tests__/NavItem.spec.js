import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";

import NavItem from "../NavItem";

it("Test of <NavItem />", ()=>{
    const mockProps = {
        title: "Title",
        link: "/"
    }
    const component = create(
        <MemoryRouter initialEntries={["/"]}>
            <NavItem {...mockProps}/>
        </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
});