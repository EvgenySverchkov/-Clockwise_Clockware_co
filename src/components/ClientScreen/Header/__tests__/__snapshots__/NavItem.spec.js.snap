// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <NavItem /> 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    href="/"
    onClick={[Function]}
  >
    Title
  </a>
</li>
`;
